ros/angel_utils/nodes/simple_2d_detection_overlay.cxx

```diff
@@ -406,6 +423,129 @@ ::collect_detections( ObjectDetection2dSet::SharedPtr const det_set )
   ++m_detset_count;
 }
 
+// ----------------------------------------------------------------------------
+void
+Simple2dDetectionOverlay
+::collect_joints( HandJointPosesUpdate::SharedPtr const joints_msg )
+{
+  auto log = this->get_logger();
+
+  // lookup image for joints_msg->source_stamp
+  // check that detection header frame_id matches
+  size_t source_nanosec_key =
+    time_key_from_header( joints_msg->source_stamp );
+  RCLCPP_DEBUG( log, "Joint source key: %zu", source_nanosec_key );
+
+  // Check that we have the image in our history that matches our joints
+  // received.
+  auto find_it = m_frame_map.find( source_nanosec_key );
+  if( find_it == m_frame_map.end() )
+  {
+    auto history_min_key = m_frame_map.begin()->first;
+    RCLCPP_WARN(
+      log,
+      "Failed to find an image in our history that matches received joints. "
+      "joint source (%zu) < (%zu) history min.",
+      source_nanosec_key, history_min_key
+      );
+    return;
+  }
+
+  // Found something. Double-checking frame source is set to the same thing?
+  if( joints_msg->header.frame_id != find_it->second->header.frame_id )
+  {
+    RCLCPP_WARN(
+      log,
+      "Received joints frame-id does not match the aligned image in "
+      "our history? d-set (%s) != (%s) image",
+      joints_msg->header.frame_id.c_str(),
+      find_it->second->header.frame_id.c_str()
+      );
+    return;
+  }
+
+  // Only plot the patient skeleton
+  if(joints_msg->hand != "patient")
+  {
+    return;
```

Comment:
Maybe add a warning message for here as well. E.g. there should only be "patient" joints travelling across the topic we utilize for this, so I imagine that other joints being on this topic would be an issue we want to be warned about.